packages/synapse-react-client/src/synapse-queries/evaluation/useEvaluation.ts

```diff
@@ -17,14 +35,26 @@ export function useGetEvaluationsInfinite(
 ) {
   const LIMIT = 20
   const { accessToken, keyFactory } = useSynapseContext()
+  const queryClient = useQueryClient()
 
   return useInfiniteQuery<PaginatedResults<Evaluation>, SynapseClientError>(
     keyFactory.getEvaluationsQueryKey(request),
-    async context =>
-      SynapseClient.getEvaluations(
+    async context => {
+      const responsePage = await SynapseClient.getEvaluations(
         { ...request, limit: LIMIT, offset: context.pageParam },
         accessToken,
-      ),
+      )
+
+      // Update the cache with each individual evaluation
+      responsePage.results.forEach(evaluation => {
+        queryClient.setQueryData(
+          keyFactory.getEvaluationByIdQueryKey(evaluation.id!),
+          evaluation,
+        )
+      })
```

Comment:
This can prevent a lot of unnecessary requests where many evaluations in your "Selected" list may already be in the cache because it was in the list of all evaluations when you were making your selection